unittest/src/kernel/mod.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{ExpectedSyscall, SyscallLogEntry};
+use std::cell::Cell;
+
+// TODO: Add Allow.
+// TODO: Add Command.
+// TODO: Add Exit.
+// TODO: Add Memop.
+// TODO: Add Subscribe.
+mod raw_syscalls_impl;
+mod thread_local;
+// TODO: Add Yield.
+
+/// A fake implementation of the Tock kernel. Provides
+/// `libtock_platform::Syscalls` by implementing
+/// `libtock_platform::RawSyscalls`. Allows `fake::Driver`s to be attached, and
+/// routes system calls to the correct fake driver.
+///
+/// Note that there can only be one `Kernel` instance per thread, as a
+/// thread-local variable is used to implement `libtock_platform::RawSyscalls`.
+/// As such, test code is given a `Rc<Kernel>` rather than a `Kernel` instance
+/// directly. Because `Rc` is a shared reference, Kernel extensively uses
+/// internal mutability.
+// TODO: Define the `fake::Driver` trait and add support for fake drivers in
+// Kernel.
+pub struct Kernel {
+    expected_syscalls: Cell<std::collections::VecDeque<ExpectedSyscall>>,
```

Comment:
Is there a reason for the fully qualified name here...?

Reply:
I generally don't add `use` statements unless those statements shorten a line is wrapping (>100 columns).